dash/include/dash/Meta.h

```diff
@@ -90,6 +90,19 @@ struct is_container_compatible :
          >
 { };
 
+/**
+ * Type trait indicating whether a type can be used for global atomic
+ * operations.
+ */
+template <typename T>
+struct is_atomic_compatible
+: public std::integral_constant<
+           bool,
+              dash::is_container_compatible<T>::value
+           && sizeof(T) <= sizeof(std::size_t)
+         >
```

Comment:
No, why? Type punning is only used when required, it's not used for doubles.
And if the user specifies a composite type for atomics, it's obvious that fetch-and-add is nonsensical.

This trait states, as decumented in it's doxygen, whether a type is eligible for atomic operations. `struct { char; char; }` is.
It does not express semantic stability of **any** atomic operation. That would be a different trait.